pm_kext/src/pm_utils.c

```diff
@@ -8,51 +8,176 @@
  *  Scope:       Kernelmode
  */
 
-#include <wdm.h>
-#include <windef.h>
-#include "pm_common.h"
-#include "pm_debug.h"
 #include "pm_utils.h"
+#include "pm_debug.h"
 
 /*
  * PORTMASTER malloc/free.
  */
-static POOL_TYPE non_paged_pool = NonPagedPool;
 
-PVOID portmaster_malloc(SIZE_T size, BOOL paged) {
-    void * pv;
-    POOL_TYPE pool = (paged? PagedPool: non_paged_pool);
+void *portmasterMalloc(size_t size, bool paged) {
+    POOL_TYPE poolFlag = (paged ? POOL_FLAG_PAGED : POOL_FLAG_NON_PAGED);
     if (size == 0) {
         return NULL;
     }
-    pv= ExAllocatePoolWithTag(pool, size, PORTMASTER_TAG);
-    if (pv != 0) {
-        RtlZeroMemory(pv, size);
-    }
+    void *pv = ExAllocatePool2(poolFlag, size, PORTMASTER_TAG); // Memory is zero initialized unless POOL_FLAG_UNINITIALIZED is specified.
```

Comment:
I replace the old `ExAllocatePoolWithTag` because it was deprecated
`ExAllocatePool2` should work the same

https://docs.microsoft.com/en-us/windows-hardware/drivers/kernel/updating-deprecated-exallocatepool-calls#exallocatepoolexallocatepoolwithtag